### ISSUE TITLE
Vim mode search should act like non-Vim mode when not triggered with `/`

### ISSUE THREAD

**Issue by shinebayar-g** [zed-industries/zed#7692]
### Summary

When searching with `/` in Vim mode, hitting `enter` moves the cursor to the next line in the buffer being searched, matching Vim's behavior. This is at odds with how `enter` behaves in a non-Vim-mode buffer search, jumping to the next result. We should refine the action here to behave like non-Vim mode unless `/` is used to start the search.

<details>
![image](https://github.com/zed-industries/zed/assets/3091558/31209340-78b2-44bf-bd49-bf19681b973c)
</details>

**Comment by mrnugget** [comment]
Hey @shinebayar-g! I cannot reproduce this. When I search in the buffer and the search input is focused, `enter` gets me to the next search result.

See here:
https://github.com/zed-industries/zed/assets/1185253/ec3eb75e-0464-43bd-b52f-03f4b6b330ca

Or do you use Vim mode by any chance?

**Comment by d1y** [comment]
I seem to have encountered the same problem (may) @mrnugget 
> The `shift-enter` key Seems like behavior is normal

```json
{
  "context": "Editor && VimControl && !VimWaiting && !menu",
  "bindings": {
    "enter": "editor::SelectLargerSyntaxNode",
    "shift-enter": "editor::SelectSmallerSyntaxNode"
  }
}
```
![dev](https://github.com/zed-industries/zed/assets/45585937/854ad8b5-bd43-46aa-a8d9-55635b1239d9)

**Comment by shinebayar-g** [comment]
> Or do you use Vim mode by any chance?

Hey @mrnugget , thanks for the hint. I noticed that It's only reproducible in Vim mode.

**Comment by mrnugget** [comment]
Okay, I see. But where does this expectation come from? As far as I know (and I just checked again): by default Vim doesn't bind `<enter>` to go-to-next-search-result. It's bound to `n` and `<cr>` is equivalent to `ctrl-m`.

**Comment by shinebayar-g** [comment]
yeah :(

**Comment by d1y** [comment]
> Okay, I see. But where does this expectation come from? As far as I know (and I just checked again): by default Vim doesn't bind `<enter>` to go-to-next-search-result. It's bound to `n` and `<cr>` is equivalent to `ctrl-m`.

And. The `shift-enter` useful in vim-mode (prev-result)? If so, it would be more intuitive to bind enter to the search input(box)

**Comment by mrnugget** [comment]
I still can't reproduce this. When I search in Vim mode with `/` and then hit `<enter>` I only jump the next line, just like in Vim.

https://github.com/zed-industries/zed/assets/1185253/b30ef37c-70ac-485a-a7a5-32353dc01dfd

**Comment by shinebayar-g** [comment]
> I still can't reproduce this. When I search in Vim mode with `/` and then hit `<enter>` I only jump the next line, just like in Vim.
> screenshot-2024-02-13-16.49.37.mp4

You beat me to it 😄 . Try the same when you're in INSERT mode.

**Comment by d1y** [comment]
> I still can't reproduce this. When I search in Vim mode with `/` and then hit `<enter>` I only jump the next line, just like in Vim.
> 
>  screenshot-2024-02-13-16.49.37.mp4

Disable all my custom keymap and use "/" + "cmd+shift+f" `<enter>` not work

> vim-mode "n" / "shift-n" prev/next result it work

![sb.gif](https://s2.loli.net/2024/02/13/lpFYaVzSTurth2b.gif)

**Comment by mrnugget** [comment]
Okay, so you want to work like in non-vim-mode when you trigger a search through `cmd+f`/`cmd-shift-f` even though you're in Vim mode, yes?

**Comment by shinebayar-g** [comment]
Yeah I would be glad if certain base functionalities work the same across vim/non-vim modes.. Like search.

**Comment by d1y** [comment]
I think that when the search-input is focused, triggering `enter` should trigger an operation (such as next-result), rather than unfocus and jump to the result. This is not in line with the habits we have developed using other applications.

**Comment by mrnugget** [comment]
> I think that when the search-input is focused, triggering `enter` should trigger an operation (such as next-result), rather than unfocus and jump to the result. This is not in line with the habits we have developed using other applications.

I personally prefer that, because it's exactly how it works in Vim: you start a search with `/`, you finish it by hitting `enter`, you jump to the next search results with `n`/`N`. That's at least the habit I've developed over a decade of using Vim :)

But I think the solution here would be to change the behaviour depending on whether the search was started via `cmd-f` or via `/`.

**Comment by d1y** [comment]
I'm just wondering why 'shift enter' works. Is this bug?

![fix](https://github.com/zed-industries/zed/assets/45585937/bd215684-2ccd-44fa-8004-723ee06b13a9)

The default bind
<img width="500" alt="image" src="https://github.com/zed-industries/zed/assets/45585937/c1a468d0-dd05-4898-a054-44d6fa5d9fc5">

**Comment by viveksjain** [comment]
Still an issue

**Comment by shinebayar-g** [comment]
Can this issue be re-evaluated? Without enter working, ctrl+f is kind of useless. Whenever I press enter it just adds a new line instead of jumping to the next match.

Clicking `n` to navigate to the next match works only if the find bar is not in focus. So UX around is not so great.

**Comment by viveksjain** [comment]
Still an issue
